### PR TITLE
Added new api for compute: GetPasswordData

### PR DESCRIPTION
**Pull request by punituee**
GetPasswordData: this expects a encrypted password returned from server
which will be decrypted with the private key

### CODE REVIEW

File: src/client/common.py [hunk 1]
```diff
@@ -1,19 +1,16 @@
-import base64
-import time
-import hmac
-import os
-import hashlib
-import six
-import sys
+import base64, time, hmac, os, hashlib
```
Comment (anantkaushik89) [inline src/client/common.py]: this a less widely followed style for importing, i would prefer sticking with the former one.

File: src/client/common.py [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
 import json
 import pprint
 import requests
-from six.moves import urllib
 import urllib as ul
 import xmltodict
 
 import exceptions
+import binascii, getpass
+from Crypto.PublicKey import RSA
```
Comment (anantkaushik89) [inline src/client/common.py:19]: Have we ensured this library would always be present in the system? And this doesnt have version dependencies.

File: src/client/common.py [hunk 3]
```diff
@@ -39,6 +36,59 @@
...
+    # want more digits than we need (hence the ternary-ish 'or')
+    width += 8 - ((width % 8) or 8)
+
+    # format width specifier: four (4) bits per hex digit
+    fmt = '%%0%dx' % (width // 4)
+
+    # prepend zero (0) to the width, to zero-pad the output
+    s = binascii.unhexlify(fmt % val)
+
+    if endianness == 'little':
```
Comment (anantkaushik89) [inline src/client/common.py]: Better to take endianness from some system library function else this wont be of much use.

File: src/client/common.py [hunk 4]
```diff
@@ -39,6 +36,59 @@
...
+        # see http://stackoverflow.com/a/931095/309233
+        s = s[::-1]
+
+    return s
+
+def decryptPassword(rsaKey, password):
+    #Undo the whatever-they-do to the ciphertext to get the integer
```
Comment (anantkaushik89) [inline src/client/common.py]: Not critical, but can remove the comments which came with the code paste.

File: src/client/common.py [hunk 5]
```diff
@@ -397,7 +468,12 @@ def curlify(service, req_str, gnucli=False, execute=False, prettyprint=True):
 
     if execute:
         request_string = requestify(service_url, params, verb)
-
+        if windowsEncryptedPassword:
+            password_json = do_request(verb, request_string, to_print=False)
```
Comment (anantkaushik89) [inline src/client/common.py:457]: A try catch block would be good. Else if the backend response changes, the client would throw weird responses.

File: src/client/common.py (hunk 2 above)
Comment (punituee) [inline src/client/common.py:19]: added it to the requirements.txt

File: src/client/common.py (hunk 5 above)
Comment (punituee) [inline src/client/common.py:457]: the response is handled in do_request. We can ignore the need of try catch here.

**Pull request merged**